### PR TITLE
Use mpiexec to run unit tests

### PR DESCRIPTION
**Pull request by quantumsteve**

### CODE REVIEW

**Comment by quantumsteve** [comment]
test this please

**Comment by quantumsteve** [comment]
Test this please

**Comment by quantumsteve** [comment]
Test this please

**Comment by idigs** [comment]
Test this please

**Comment by idigs** [comment]
Test this please

**Comment by idigs** [comment]
Test this please

**Comment by quantumsteve** [comment]
Test this please

**Comment by quantumsteve** [comment]
@cianciosa `mars_test` passes on one process but is failing failing on two or more processes. Are there random number initialization differrences?

**Comment by mbeidler3** [comment]
@quantumsteve I've been trying to create gold data for multiple processor runs. However, when running on my machine, each processor has rank=0 and nmpi=1. Also, it's even more confusing because only one processor prints the output.korc file even though that task is done by rank=0. I compiled the recent version on Perlmutter using their module system (cray-mpich for MPI, rather than openmpi on my system) and it initialized the MPI processes as expected.

**Comment by mbeidler3** [comment]
@quantumsteve, I'm getting the same result for a build with mpich. On my machine, I see it compiling with gfortran and including the MPI library directoy, while on Perlmutter I see it compiling with the wrapper compiler ftn. Could the issue be that my system is not compiling with the correct MPI-wrapped compilers?

**Comment by quantumsteve** [comment]
> @quantumsteve, I'm getting the same result for a build with mpich. On my machine, I see it compiling with gfortran and including the MPI library directoy, while on Perlmutter I see it compiling with the wrapper compiler ftn. Could the issue be that my system is not compiling with the correct MPI-wrapped compilers?

Something is going on with the wrapper. If I add `export FC=mpifort` to build.sh, I get one output file per process. Not setting `FC` and I only get a single file for ranks 1, 2, and 4. With rank 8, I get 8 files in both cases :confused:

**Comment by quantumsteve** [comment]
Test this please

File: test/CMakeLists.txt
Comment (idigs) [inline test/CMakeLists.txt:1]: As discussed today, you have broken out mars_test to only use rank_1

**Review by idigs — APPROVED** [review]
I approve these changes

**Pull request merged**